src/Monolog/LogRecord.php

```diff
@@ -0,0 +1,122 @@
+<?php declare(strict_types=1);
+
+/*
+ * This file is part of the Monolog package.
+ *
+ * (c) Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Monolog;
+
+use ArrayAccess;
+
+/**
+ * Monolog log record
+ *
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ * @template-implements \ArrayAccess<'message'|'level'|'context'|'level_name'|'channel'|'datetime'|'extra', int|string|\DateTimeImmutable|array<mixed>>
+ * @phpstan-import-type Level from Logger
+ * @phpstan-import-type LevelName from Logger
+ */
+class LogRecord implements \ArrayAccess
+{
+    private const MODIFIABLE_FIELDS = [
+        'extra' => true,
+        'formatted' => true,
+    ];
+
+    /** @var 'DEBUG'|'INFO'|'NOTICE'|'WARNING'|'ERROR'|'CRITICAL'|'ALERT'|'EMERGENCY' */
+    public readonly string $levelName; // TODO enum?
+
+    public function __construct(
+        public readonly \DateTimeImmutable $datetime,
+        public readonly string $channel,
+        /** @var Logger::DEBUG|Logger::INFO|Logger::NOTICE|Logger::WARNING|Logger::ERROR|Logger::CRITICAL|Logger::ALERT|Logger::EMERGENCY */
+        public readonly int $level, // TODO enum?
```

Comment:
Yes, let's make this en enum while we're at it.

Reply:
The catch with enum is it means one cannot extend the allowed log levels anymore. On the other hand, extending that is probably a bad idea, and I am not sure if anyone actually does it.